README.md

```diff
@@ -57,25 +57,28 @@ query Invoices{
 }
 ```
 
-A successful response is a JSON object.
+It will validate the query based on current schema. In case of validation errors or any other connection related issues you'll get `Graphlient::Errors::Client` describing the error.
 
-On failure the client raises `Graphlient::Errors::HTTP` which contains the `response` object.
+On success you get a response object and you can access `data` and iterate over results. In example above we can access first line item's price by:
+```ruby
+response.data.invoice&.line_items.first&.price
```

Comment:
I would remove `&`, that's beyond the point.

Reply:
obviously i've been to excited about `&.`